### PR TITLE
mem-ruby: allow comparison of int and Addr in SLICC

### PR DESCRIPTION
**Pull request by arteen1000**
allow easy isolation of specific addresses in coherence protocols. useful for debugging.

Change-Id: I93e07956b8e29837219d328dacfbd5c6067c1a62

### CODE REVIEW

File: src/mem/slicc/ast/OperatorExprAST.py [hunk 1]
```diff
@@ -62,7 +62,15 @@ def generate(self, code, **kwargs):
         # Figure out what the input and output types should be
         if self.op in ("==", "!=", ">=", "<=", ">", "<"):
             output = "bool"
-            if ltype != rtype:
+
+            if (
```
Comment (mattsinc) [inline src/mem/slicc/ast/OperatorExprAST.py:66]: I don't think the formatting of this if matches the formatting elsewhere in the file

**Review by mattsinc — CHANGES_REQUESTED** [review]
Beyond the one minor change, I agree with the spirit of this change.  @powerjg should view too though -- I think it's fine to allow int and Addr in SLICC, but I guess it's worth us discussing what the purpose of allow these return types is from SLICC?

**Review by powerjg — APPROVED** [review]
Fine with me :)

**Comment by powerjg** [comment]
@mattsinc can you re-review since you "requested changes" to allow this to merge?

**Review by mattsinc — APPROVED** [review]

**Pull request merged**